lib/selinux/selinux_linux.go

```diff
@@ -0,0 +1,179 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package selinux
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"io/fs"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/gravitational/trace"
+	ocselinux "github.com/opencontainers/selinux/go-selinux"
+)
+
+const (
+	selinuxConfig        = "/etc/selinux/config"
+	selinuxTypeTag       = "SELINUXTYPE"
+	permissiveModuleName = "permissive_" + domain
+)
+
+// Copyright 2025 Open Container Initiative
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Original source: https://github.com/opencontainers/selinux/blob/main/go-selinux/selinux_linux.go#L229
+// Modified to return an error.
+func readConfig(target string) (string, error) {
+	in, err := os.Open(selinuxConfig)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+	defer in.Close()
```

Comment:
If we're not expecting this file to be huge we could use `os.ReadFile` and split the parsing and the I/O in two different functions.